lib/kitchen/driver/gce.rb

```diff
@@ -141,26 +141,97 @@ def destroy(state)
         state.delete(:zone)
       end
 
+      def old_disk_configuration_present?
+        !config[:autodelete_disk].nil? || !config[:disk_size].nil? || !config[:disk_type].nil?
+      end
+
+      def new_disk_configuration_present?
+        !config[:disks].nil?
+      end
+
+      def create_disks_config
+        # This can't be present in default_config because we couldn't
+        # determine which disk configuration the user used otherwise
+        disk_default_config = {
+          autodelete_disk: true,
+          disk_size: 10,
+          disk_type: "pd-standard",
+        }
+
+        if old_disk_configuration_present?
+          # If the old disk configuration is used,
+          # we'll convert it to the new one
+          config[:disks] = {}
+          config[:disks][server_name] = {
+            boot: true,
+          }
+          config[:disks][server_name][:autodelete_disk] = config[:autodelete_disk].nil? ? disk_default_config[:autodelete_disk] : config[:autodelete_disk]
+          config[:disks][server_name][:disk_size] = config[:disk_size].nil? ? disk_default_config[:disk_size] : config[:disk_size]
+          config[:disks][server_name][:disk_type] = config[:disk_type].nil? ? disk_default_config[:disk_type] : config[:disk_type]
```

Comment:
Style recommendation:

The creation of a new_disk_configuration based on an old_disk_configuration can be simplified and made maybe a little easier to read with the [`Hash#fetch`](https://ruby-doc.org/core-2.5.0/Hash.html#method-i-fetch) method.

```ruby
# If the old disk configuration is used,
# we'll convert it to the new one
config[:disks] = {
  disk1: {
    boot: true,
    autodelete_disk: config.fetch(:autodelete_disk, disk_default_config[:autodelete_disk]),
    disk_size: config.fetch(:disk_size, disk_default_config[:disk_size]),
    disk_type: config.fetch(:disk_type, disk_default_config[:disk_type]),
  }
}
```